client/app/scripts/superdesk/editor/editor.js

```diff
@@ -199,11 +203,125 @@ function FindReplaceCommand(rootNode) {
     }
 }
 
+function SpellcheckService($q) {
+
+    var selection = document.getSelection();
+
+    this.check = function check(text) {
+        var words = _.uniq(_.filter(text.split(/[\W0-9]+/)));
```

Comment:
This will not always work correctly, because `\W` does not recognize non-ascii letters. In the AES project we solved this problem by using [XRegExp](http://xregexp.com/). We used the following expression to search for words in a text:

```javascript
var unicodeWords = new XRegExp('(\\p{Letter}|\\d)+', 'g');
```
Hope this helps!